### PR TITLE
Move *.h.cmake files into cMake/header/ folder

### PR DESCRIPTION
**Pull request by mosfet80**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7350">pipeline status</a> for feature branch [PR_7350](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7350). Pipeline [609183489](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/609183489) was triggered at [3d9e6c6](https://github.com/FreeCAD/FreeCAD/pull/7350/commits/3d9e6c6a7072268382adabdddb1da7facd054963). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by yorikvanhavre** [comment]
Rename header -> headers maybe?

**Comment by mosfet80** [comment]
Done

**Comment by luzpaz** [comment]
This may screw-up some downstream packager scripts, jfyi

**Comment by chennes** [comment]
@luzpaz can you invite someone from that community to take a look at this then? I don't see any particular issues.

**Comment by luzpaz** [comment]
I'll ping some packagers soon. Keeping this ticket open in a tab :rofl:

**Comment by luzpaz** [comment]
Heads up FreeCAD packagers/maintainers :point_up: 

| Packager                     | Platform/Dist      |
|:-----------------------------|:------------------:|
| @xyproto                     | Arch/AUR           |
| @adrianinsaval               | Arch/AUR           |
| @looooo                      | Conda              |
| @kkremitzki                  | Debian/PPA         |
| @hobbes1069                  | Fedora             |
| @PrzemoF                     | Fedora Copr        |
| @waebbl                      | gentoo             |
| @adrianschroeter             | openSUSE           |
| @ipatch                      | macOS homebrew     |
| @ppd                         | Snap               |
| @hfiguiere                   | flatpak            |
| @donovaly                    | Windows LibPack    |
| @apeltauer                   | Windows LibPack    |
| @sgrogan                     | Windows            |
| @moench-tegeder              | FreeBSD            |
| @mandian                     | OpenMandriva       |

**Comment by xyproto** [comment]
@luzpaz Thanks for the heads up!

Small correction: I don't maintain any FreeCAD-related packages in the unofficial AUR, just the official one.

File: cMake/FreeCAD_Helpers/SetupSalomeSMESH.cmake [hunk 1]
```diff
@@ -130,7 +130,7 @@ macro(SetupSalomeSMESH)
         endif()
 
         set(SMESH_FOUND TRUE)
-        configure_file(SMESH_Version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/SMESH_Version.h)
+        configure_file(${CMAKE_SOURCE_DIR}/cMake/header/SMESH_Version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/SMESH_Version.h)
```
Comment (waebbl) [inline cMake/FreeCAD_Helpers/SetupSalomeSMESH.cmake:133]: A typo here still fails the CI: `s|cMake/header|cMake/headers|`

**Comment by hobbes1069** [comment]
I don't have any issues with the concept.

File: cMake/FreeCAD_Helpers/SetupSalomeSMESH.cmake (hunk 1 above)
Comment (luzpaz) [reply cMake/FreeCAD_Helpers/SetupSalomeSMESH.cmake:133]: @mosfet80 :point_up:

**Comment by donovaly** [comment]
@mosfet80 , Ping.

**Comment by xyproto** [comment]
FreeCAD from `master` appears to build and run as it should here.

**Pull request closed without merging**